programs/ssl/ssl_server2.c

```diff
@@ -2377,8 +2356,31 @@ int main( int argc, char *argv[] )
     mbedtls_ssl_cookie_init( &cookie_ctx );
 #endif
 
-
-
+    if( ssl         == NULL ||
```

Comment:
Blocker: I think this block comes too late. We've already passed each of these pointers to some `xxx_init()` function that happily dereferenced it without checking if it was NULL. I suggest we swap the two blocks.

(I checked the client and the order looks correct there.)